File: src/client/java/fr/hardel/asset_editor/client/javafx/components/layout/editor/NoPermissionPage.java
package fr.hardel.asset_editor.client.javafx.components.layout.editor;

import fr.hardel.asset_editor.AssetEditor;
import fr.hardel.asset_editor.client.javafx.VoxelColors;
import fr.hardel.asset_editor.client.javafx.components.ui.SvgIcon;
import javafx.geometry.Pos;
import javafx.scene.control.Label;
import javafx.scene.layout.VBox;
import net.minecraft.client.resources.language.I18n;
import net.minecraft.resources.Identifier;

public final class NoPermissionPage extends VBox {

    private static final Identifier LOCK_ICON = Identifier.fromNamespaceAndPath(AssetEditor.MOD_ID, "icons/lock.svg");

    public NoPermissionPage() {
        setAlignment(Pos.CENTER);
        setSpacing(16);
        getStyleClass().add("studio-root");

        SvgIcon icon = new SvgIcon(LOCK_ICON, 48, VoxelColors.ZINC_500);
        icon.setOpacity(0.5);

        Label title = new Label(I18n.get("studio:permission.no_access.title"));
        title.setStyle("-fx-text-fill: #a1a1aa; -fx-font-size: 20; -fx-font-weight: 700;");

        Label description = new Label(I18n.get("studio:permission.no_access.description"));
        description.setStyle("-fx-text-fill: #71717a; -fx-font-size: 14;");
        description.setWrapText(true);
        description.setMaxWidth(400);
        description.setAlignment(Pos.CENTER);

        getChildren().addAll(icon, title, description);
    }
}

File: src/client/kotlin/fr/hardel/asset_editor/client/compose/components/layout/editor/NoPermissionPage.kt
package fr.hardel.asset_editor.client.compose.components.layout.editor

import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.material.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import fr.hardel.asset_editor.AssetEditor
import fr.hardel.asset_editor.client.compose.StudioColors
import fr.hardel.asset_editor.client.compose.StudioTypography
import fr.hardel.asset_editor.client.compose.components.ui.SvgIcon
import net.minecraft.client.resources.language.I18n
import net.minecraft.resources.Identifier

private val LOCK_ICON = Identifier.fromNamespaceAndPath(AssetEditor.MOD_ID, "icons/lock.svg")

@Composable
fun NoPermissionPage(modifier: Modifier = Modifier) {
    // Compose-only: état vide d'autorisation, sans équivalent TSX direct dans le layout editor.
    Box(
        contentAlignment = Alignment.Center,
        modifier = modifier.fillMaxSize()
    ) {
        // div: flex flex-col items-center gap-4
        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            Box(modifier = Modifier.alpha(0.5f)) {
                SvgIcon(location = LOCK_ICON, size = 48.dp, tint = StudioColors.Zinc500)
            }

            Text(
                text = I18n.get("studio:permission.no_access.title"),
                style = StudioTypography.bold(20),
                color = StudioColors.Zinc400
            )

            Text(
                text = I18n.get("studio:permission.no_access.description"),
                style = StudioTypography.regular(14),
                color = StudioColors.Zinc600,
                textAlign = TextAlign.Center
            )
        }
    }
}
